src/main/resources/assets/js/app/settings/wizard/ProjectReadAccessWizardStepForm.ts

```diff
@@ -14,47 +14,112 @@ import {ProjectReadAccess, ProjectReadAccessType} from '../data/project/ProjectR
 import {PrincipalKey} from 'lib-admin-ui/security/PrincipalKey';
 import {Principal} from 'lib-admin-ui/security/Principal';
 import {GetPrincipalsByKeysRequest} from 'lib-admin-ui/security/GetPrincipalsByKeysRequest';
-import {DefaultErrorHandler} from 'lib-admin-ui/DefaultErrorHandler';
 import {ProjectPermissions} from '../data/project/ProjectPermissions';
 import {ValidationRecording} from 'lib-admin-ui/form/ValidationRecording';
+import {LocaleComboBox} from 'lib-admin-ui/ui/locale/LocaleComboBox';
+import {Locale} from 'lib-admin-ui/locale/Locale';
+import {LocaleLoader} from 'lib-admin-ui/locale/LocaleLoader';
 
 export class ProjectReadAccessWizardStepForm
     extends SettingDataItemWizardStepForm<ProjectViewItem> {
 
-    private readAccessRadioGroup: RadioGroup;
+    private readAccessRadioGroup?: RadioGroup;
 
-    private readAccessRadioGroupFormItem: FormItem;
+    private readAccessRadioGroupFormItem?: FormItem;
 
-    private principalsCombobox: PrincipalComboBox;
+    private principalsCombobox?: PrincipalComboBox;
+
+    private localeCombobox: LocaleComboBox;
 
     constructor() {
         super();
     }
 
-    layout(item: ProjectViewItem) {
+    layout(item: ProjectViewItem): Q.Promise<void> {
         if (!item) {
-            return;
+            return Q(null);
         }
 
-        const readAccess: ProjectReadAccess = item.getData().getReadAccess();
-        this.readAccessRadioGroup.setValue(readAccess.getType(), true);
+        const layoutPromises: Q.Promise<any>[] = [];
+
+        layoutPromises.push(this.layoutLanguage(item.getLanguage()));
+
+        if (item.isDefaultProject()) {
+            return Q.all(layoutPromises).spread<void>(() => Q<void>(null));
+        }
+
+        layoutPromises.push(this.layoutReadAccess(item.getReadAccess(), item.getPermissions()));
 
-        this.updateFilteredPrincipalsByPermissions(item.getPermissions());
+        return Q.all(layoutPromises).spread<void>(() => Q<void>(null));
```

Comment:
This would probably be more readable with one `return`:

```
if (!item.isDefaultProject()) {
  layoutPromises.push(this.layoutReadAccess(item.getReadAccess(), item.getPermissions()));
}        

return Q.all(layoutPromises).spread<void>(() => Q<void>(null));

```